zero_functional.nim

```diff
@@ -374,9 +406,7 @@ proc inlineFold(ext: ExtNimNode): ExtNimNode{.compileTime.} =
 proc inlineSeq(ext: ExtNimNode): ExtNimNode {.compileTime.} =
   let itIdent = ext.itNode()
   let node = ext.node
-  let idxIdent = newIdentNode(indexVariableName)
   ext.node = quote:
-    var `idxIdent` = 0
```

Comment:
define the idx variable outside of the loop - otherwise the `compileTimeTypeInfer` will not work when using the idx in some combinations with the sequence functions